test/token_utils.spec.js

```diff
@@ -0,0 +1,101 @@
+/* globals describe, it, expect */
+
+"use strict";
+
+const util = require('util');
+
+let tokenUtils = require('../lib/token_utils');
+
+describe('Errors reported with wrong parameters', () => {
+  let bad_params = [
+    null,
+    undefined,
+    1
+  ];
+  bad_params.forEach( item => {
+    it(`throws with wrong parameter type '${item}' `, () => {
```

Comment:
These tests for `parseToken` have the same test names as the below tests for `formatTokenForHeader`; add function name to help differentiate between them?